File: AgentPlanner.Services/PaymentMethodService.cs
using AgentPlanner.Entities.Client;
using AgentPlanner.Repositories;
using AgentPlanner.Entities.Mappers;

namespace AgentPlanner.Services
{
    public class PaymentMethodService
    {
        private readonly PaymentMethodRepository _paymentRepository;

        public PaymentMethodService()
        {
            _paymentRepository = new PaymentMethodRepository();
        }
        public PaymentMethod[] GetPaymentMethods()
        {
            return _paymentRepository.GetAll().ToDtos();
        }

        public PaymentMethod GetPaymentMethod(int paymentMethodId)
        {
            return _paymentRepository.Get(paymentMethodId).ToDto();
        }
    }
}

File: Published/App/Services/paymentMethodService.js
var app = angular.module('agent_planner');

app.service('paymentMethodService', [
    'http', function (http) {
        return {
            getPaymentMethods: function (onSuccess) {
                http.get("/api/paymentMethod/list/", {}, onSuccess);
            },
            get: function(paymentMethodId, onSuccess) {
                http.get("/api/paymentMethod/" + paymentMethodId, {}, onSuccess);
            }
        }
    }
]);